common/account_utils/src/lib.rs

```diff
@@ -0,0 +1,129 @@
+//! Provides functions that are used for key/account management across multiple crates in the
+//! Lighthouse project.
+
+use eth2_keystore::Keystore;
+use eth2_wallet::Wallet;
+use rand::{distributions::Alphanumeric, Rng};
+use std::fs::{self, File};
+use std::io;
+use std::io::prelude::*;
+use std::os::unix::fs::PermissionsExt;
+use std::path::{Path, PathBuf};
+
+pub use eth2_wallet::PlainText;
+
+/// The `Alphanumeric` crate only generates a-z, A-Z, 0-9, therefore it has a range of 62
+/// characters.
+///
+/// 62**48 is greater than 255**32, therefore this password has more bits of entropy than a byte
+/// array of length 32.
+const DEFAULT_PASSWORD_LEN: usize = 48;
+
+/// Returns the "default" path where a wallet should store its password file.
+pub fn default_wallet_password_path<P: AsRef<Path>>(wallet_name: &str, secrets_dir: P) -> PathBuf {
+    secrets_dir.as_ref().join(format!("{}.pass", wallet_name))
+}
+
+/// Returns a password for a wallet, where that password is loaded from the "default" path.
+pub fn default_wallet_password<P: AsRef<Path>>(
+    wallet: &Wallet,
+    secrets_dir: P,
+) -> Result<PlainText, io::Error> {
+    let path = default_wallet_password_path(wallet.name(), secrets_dir);
+    fs::read(path).map(|bytes| PlainText::from(strip_off_newlines(bytes)))
+}
+
+/// Returns the "default" path where a keystore should store its password file.
+pub fn default_keystore_password_path<P: AsRef<Path>>(
+    keystore: &Keystore,
+    secrets_dir: P,
+) -> PathBuf {
+    secrets_dir
+        .as_ref()
+        .join(format!("0x{}", keystore.pubkey()))
+}
+
+/// Reads a password file into a Zeroize-ing `PlainText` struct, with new-lines removed.
+pub fn read_password<P: AsRef<Path>>(path: P) -> Result<PlainText, io::Error> {
+    fs::read(path).map(strip_off_newlines).map(Into::into)
+}
+
+/// Creates a file with `600 (-rw-------)` permissions.
+pub fn create_with_600_perms<P: AsRef<Path>>(path: P, bytes: &[u8]) -> Result<(), io::Error> {
+    let path = path.as_ref();
+
+    let mut file = File::create(&path)?;
+
+    let mut perm = file.metadata()?.permissions();
+
+    perm.set_mode(0o600);
+
+    file.set_permissions(perm)?;
+
+    file.write_all(bytes)?;
+
+    Ok(())
+}
+
+/// Generates a random alphanumeric password of length `DEFAULT_PASSWORD_LEN`.
+pub fn random_password() -> PlainText {
+    rand::thread_rng()
+        .sample_iter(&Alphanumeric)
+        .take(DEFAULT_PASSWORD_LEN)
+        .collect::<String>()
+        .into_bytes()
+        .into()
+}
+
+/// Remove any number of newline or carriage returns from the end of a vector of bytes.
+pub fn strip_off_newlines(mut bytes: Vec<u8>) -> Vec<u8> {
```

Comment:
[`strip_suffix()`](https://doc.rust-lang.org/std/string/struct.String.html#method.strip_suffix) has been stabilized recently

Reply:
At the moment nothing stops a user from using *any* bytes that don't end in `/r` or `/n` and I think we'd lose that if we were to convert `bytes` into a `String`.

I think Eth2 clients are in agreement that we should strip newlines but I'm not sure there's consensus on only allowing text. If we start restricting to text-only we might not be able to import keys created by some other client. Because of this, I'm tempted to leave it as is.